### PR TITLE
Adds Category support to deep linking

### PR DESCRIPTION
**Pull request by colinta**
:metal: @rynbjyn

### CODE REVIEW

**Comment by rynbyjn** [comment]
Cool! One question, does this handle `/discover/featured`, `/discover/recommended`, `/discover`, and `/` as the same path? This is the regex we use in the webapp to determine this: `/^\/(?:discover(\/featured|\/recommended)?)?$/`

**Comment by colinta** [comment]
I'll check - probably not, and probably easy to add support for those.

**Comment by colinta** [comment]
Does now!

Supports: (w/ or w/out trailing slash)

```
/discover
/discover/featured
/discover/recommended
```

File: Sources/Networking/ElloURI.swift [hunk 1]
```diff
@@ -11,11 +11,12 @@ import Keys
 
 public enum ElloURI: String {
     // matching stream or page in app
-    case Discover = "discover"
+    case Discover = "discover(/featured|/recommended)?/?$"
```
Comment (rynbyjn) [inline Sources/Networking/ElloURI.swift]: I guess y'all don't need to support the logged out `/` being a discover stream too.

Comment (colinta) [inline Sources/Networking/ElloURI.swift]: Yeah not at the moment

Comment (colinta) [inline Sources/Networking/ElloURI.swift]: Or... do we?  Not sure where `"/"` would go, I'll check.

Comment (colinta) [inline Sources/Networking/ElloURI.swift]: It does go to discover 👍

Comment (rynbyjn) [inline Sources/Networking/ElloURI.swift]: tight :metal:

**Comment by rynbyjn** [comment]
LGTM :metal:

**Comment by colinta** [comment]
These tests are just lying, I dunno what's up: ```Executed 2001 tests, with 0 failures (0 unexpected) in 99.713 (100.641) seconds```

**Comment by steam** [comment]
👍

**Pull request merged**